docs/Infrastructure-Deployment-Scenarios-Repository-Structure.md

```diff
@@ -1,162 +1,172 @@
 should not know what product or deployment pattern is going to be deployed into
 this infrastructure. But, this raises the question: How can we provision
 an infrastructure without knowing number of VM instances required? Read on to
 the next section to find out. :-)
 
 ## Deploying products
 
-Next step is to deploy WSO2 products in the provisioned infrastructure.
-WSO2 provides a set of deployment patterns that can be used for this purpose.
+Now that the infrastructure provisioning is done, the
+next step is to deploy WSO2 products in this infrastructure.
 The products can be deployed via a configuration mgt tool like Ansible/Puppet.
 
-The product teams need to include a **`deploy.sh`** at the root of the respective
-git repository that takes care of invoking the deployment scripts. In addition,
+The product teams need to include a metadata file named **testgrid.yaml**
+that describe how to run the repo. This file (testgrid.yaml) needs to be placed
+at the root of the deployment repository. This configuration takes the following
+format:
 
+### testgrid.yaml :: deploymentConfig
 
-### WSO2Deployfile
-This is syntax of the WSO2Deployfile:
-
-```
-WSO2Deployfile:
+```yaml
+deploymentConfig:
   deploymentPatterns:
     - name: <string>
       description: <string>
       dir: <string>
-      inputsForInfrastructurefile:
-          - requiredVMCount: <int>
-          - ...
-    - ...
-    - ...
-  InputParameters:
-    - name: <string>
-      type: <Type>
-    - ...
+      scripts:
+        - type: <string>
+          description: <string>
+          file: </path/to/deploy.sh>
+          name: <string>
+          phase: <string {CREATE, DEPLOY, DESTROY}>
+          inputParameters:
 
 ```
 
-```
-Type enum {
- string,
- int,
- list
-}
-```
-
-Example WSO2Deployfile:
-```
-WSO2Deployfile:
+Example testgrid.yaml for deployment repos:
+```yaml
+deploymentConfig:
   deploymentPatterns:
-    - name: 01-two-node-deployment
-      description: Deploys a two node IS cluster
-      dir: patterns/01-two-node-deployment
-      inputsForInfrastructurefile:
-          - requiredVMCount: 2
-  InputParameters:
-    - name : hostNames
-      type : <list>
-    - name : sshKey
-      type : <string>
+    - name: 01-testgrid-is-deployment
+      description: Deploys an IS node in AWS
+      dir: .
+      scripts:
+        - type: SHELL
+          file: "deploy.sh"
+          name: "deploy"
+          phase: CREATE
 ```
 
-Note: the metadata under _inputsForInfrastructurefile_ element are fed into
-the infrastructure provisioning script. This needs to be done because even though
-the infrastructure and WSO2 product deployment are orthogonal, there is some-level
-of coupling between the two. One good example is the number of VM machines required
-for a given deployment pattern. But, this should not, however, specify any
-infrastructure _provider_ specific parameters.
+<!-- Note: the metadata under _inputsForInfrastructurefile_ element are fed into -->
```

Comment:
By making these into comments, these lines are invisible when viewing this file.
Is that the intention or do we need to show these as "comments" in the file? :)

Reply:
Yes. This feature is no longer there. This will only come in future. Didn't want to expose this without really validating the usecase.